### PR TITLE
[clang] Fix static analyzer concerns in #embed code

### PR DESCRIPTION
**Pull request by Fznamznon**
1. Dead code in `LookupEmbedFile`. The loop always exited on the first iteration. This was also causing a bug of not checking all directories provided by `--embed-dir`.

2. Use of uninitialized variable `CurTok` in `LexEmbedParameters`. It was used to initialize the field which seems to be unused. Removed unused field, this way `CurTok` should be initialized by Lex method.

### CODE REVIEW

File: clang/test/Preprocessor/embed_file_not_found_chevron.c [hunk 1]
```diff
@@ -1,4 +1,5 @@
-// RUN: %clang_cc1 -std=c23 %s -E -verify
+// RUN: %clang_cc1 -std=c23 %s -E -verify --embed-dir=%S --embed-dir=%S/Inputs
 
+#embed <jk.txt>
```
Comment (AaronBallman) [inline clang/test/Preprocessor/embed_file_not_found_chevron.c:3]: It feels like this test is unrelated to the test name now; should we move this into its own test file named something like `embed_search_paths.c` or something?

**Review by AaronBallman — APPROVED** [review]
LGTM but a question about whether a test should be split off or not.

File: clang/test/Preprocessor/embed_file_not_found_chevron.c (hunk 1 above)
Comment (Fznamznon) [reply clang/test/Preprocessor/embed_file_not_found_chevron.c:3]: Makes sense.

**Review by AaronBallman — APPROVED** [review]
LGTM!
